### PR TITLE
18940 Add new draft type for amalgamation

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:*
https://github.com/bcgov/entity/issues/18940

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/src/auth_api/services/affiliation.py [hunk 1]
```diff
@@ -457,6 +457,10 @@ def _combine_affiliation_details(details):
             if 'nrNumber' in business and (nr_num := business['nrNumber']):
                 if business['nrNumber'] in name_requests:
                     business['nameRequest'] = name_requests[nr_num]['nameRequest']
+                    # Update the draft type if the draft NR request is for amalgamation
+                    if business.get('draftType', None) \
```
Comment (seeker25) [inline auth-api/src/auth_api/services/affiliation.py:461]: Should we throw in a unit test for this? Not sure if any exist

**Review by seeker25 — APPROVED** [review]

File: auth-api/src/auth_api/services/affiliation.py [hunk 2]
```diff
@@ -457,6 +457,10 @@ def _combine_affiliation_details(details):
...
+                            and business['nameRequest']['request_action_cd'] == NRActionCodes.AMALGAMATE.value:
+                        business['draftType'] = CorpType.ATMP.value
```
Comment (severinbeauvais) [inline auth-api/src/auth_api/services/affiliation.py:463]: I expected to see this code alongside other code that sets TMP or RTMP ...?

File: auth-api/src/auth_api/utils/enums.py [hunk 3]
```diff
@@ -353,3 +354,21 @@ class NotificationTypes(Enum):
...
+    CONVERSION = 'CNV'  # aka Alteration
+    DBA = 'DBA'  # doing business as
+    MOVE = 'MVE'  # continuation in
+    NEW_BUSINESS = 'NEW'  # incorporate or register
+    RESTORE = 'REH'  # restore or reinstate
+    RENEW = 'REN'  # restore with new name request
+    RESTORATION = 'REST'  # FUTURE: unused? delete?
+    RESUBMIT = 'RESUBMIT'  # FUTURE: unused? delete?
+
+    INFO = 'INFO'  # special value for sub-menu
```
Comment (severinbeauvais) [inline auth-api/src/auth_api/utils/enums.py:374]: You don't need INFO in the API; it's only for Namerequest UI.

Comment (kzdev420) [reply auth-api/src/auth_api/utils/enums.py]: Updated

File: auth-api/src/auth_api/services/affiliation.py (hunk 2 above)
Comment (kzdev420) [reply auth-api/src/auth_api/services/affiliation.py:463]: It is here line 450. `sets TMP or RTMP` but at the moment, we don't have the NameRequests.
So, I updated the draftType when combine the NR

Comment (severinbeauvais) [reply auth-api/src/auth_api/services/affiliation.py:463]: I see. I still think that AAs are fundamentally the same as IAs and Regs (even including NRs) but if you and Travis think this is OK then I'm OK with this, too.

**Pull request merged**